packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -2069,17 +2072,31 @@ export class Bucket extends BucketBase {
   }
 
   /**
-   * Allows the LogDelivery group to write, fails if ACL was set differently.
+   * Allows the 'logging.s3.amazonaws.com' service principal to write log events
+   * to this bucket.
    *
    * @see
-   * https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html
    */
-  private allowLogDelivery() {
-    if (this.accessControl && this.accessControl !== BucketAccessControl.LOG_DELIVERY_WRITE) {
-      throw new Error("Cannot enable log delivery to this bucket because the bucket's ACL has been set and can't be changed");
-    }
+  private allowLogDelivery(sourceBucket: IBucket, logFilePrefix?: string) {
+    this.grantLogDeliveryAccessUsingIamPolicy(sourceBucket, logFilePrefix);
+  }
+
+  private grantLogDeliveryAccessUsingIamPolicy(sourceBucket: IBucket, logFilePrefix?: string) {
+    const prefix = logFilePrefix ?? '*';
+
+    this.addToResourcePolicy(new PolicyStatement({
+      principals: [new ServicePrincipal('logging.s3.amazonaws.com')],
+      actions: ['s3:PutObject'],
+      conditions: {
```

Comment:
Added.

> Does the policy need to be added to the source bucket if the access logs are configured to be sent to the source bucket?

I think so. Otherwise, the service principal wouldn't have permission to write the logs.

Reply:
Not sure if I understand... today the `LOG_DELIVERY_WRITE` ACL is applied to the target bucket. If it happens that the source and target are the same, then the ACL is applied to the source bucket. By the same logic, this change applies the policy to the source bucket if source and target are the same.

Reply:
I added a comment that tries to clarify.